woocommerce/storefront
# Follow WordPress CSS coding standards

Adds `stylelint-config-wordpress` package, and update stylelint to use `stylelint-config-wordpress\scss` config which enforces the [WordPress CSS coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/).

I've disabled some rules which need substantial changes to fix and fixed all the violations reported. We can work towards re-enabling these rules one by one in the future.

Reviewing the [diff without whitespace changes](https://github.com/woocommerce/storefront/pull/1431/files?w=1) is helpful.

References:
- https://make.wordpress.org/core/handbook/best-practices/coding-standards/css/
- https://github.com/WordPress-Coding-Standards/stylelint-config-wordpress


### How to test the changes in this Pull Request:

1. Run `npm run css`. You should see no linting errors.
2. Open a WordPress installation with StoreFront enabled and check for visual regressions.

### Changelog

> Tweak – update stylelint to use `stylelint-config-wordpress\scss` config and refactor SCSS files to follow WordPress CSS coding standards.
